### PR TITLE
Isolate RFC6979 for unit testing

### PR DESCRIPTION
**Pull request by dcousens**
Based on https://github.com/bitcoinjs/tiny-secp256k1/pull/16

I don't know if this is the best way to do this,   would appreciate some opinions.
Simply for peace of mind that our original RFC6979 impl. is specifically tested (as it was previously).

### CODE REVIEW

File: ecurve.js [hunk 1]
```diff
@@ -243,7 +187,7 @@ function sign (hash, x) {
     if (s.isZero() === 0) return false
 
     return true
-  })
+  }, isPrivate)
```
Comment (dcousens) [inline ecurve.js]: had to pass this to prevent circular dependency problems ... a bit gross

File: tests/fixtures/rfc6979.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
+[
```
Comment (dcousens) [inline tests/fixtures/rfc6979.json]: our previous test fixtures https://github.com/bitcoinjs/bitcoinjs-lib/blob/d8b66641b3ae3f3f5ad6f0c04a41877da20b34ef/test/fixtures/ecdsa.json

**Comment by dcousens** [comment]
@junderw this is relevant before your PR

**Comment by dcousens** [comment]
@junderw any opinions on the approach token?

File: ecurve.js (hunk 1 above)
Comment (dcousens) [reply ecurve.js]: @junderw specifically here

**Comment by junderw** [comment]
Nope. This is pretty good. We are able to test that the loop function works.

tbh, 0 and 1 are the only ones we really need to test... not sure why I made fixtures for 5, 15, etc. lol (These fixtures are the tests I made when implementing RFC6979 for coinbin.)

But it doesn't take too long so as is is fine.

**Comment by dcousens** [comment]
@junderw it doens't hurt to verify the internal consistency with up to `k:15` IMHO.

**Pull request merged**